adapter/internal/messaging/notification_listener.go

```diff
@@ -282,16 +287,16 @@ func handleApplicationEvents(data []byte, eventType string) {
 		}
 
 		if applicationEvent.Event.Type == applicationCreate {
-			eh.AppList.List = append(eh.AppList.List, application)
-			logger.LoggerInternalMsg.Infof("Application %s is added.", applicationEvent.ApplicationName)
+			eh.ApplicationMap[application.UUID] = &application
+			logger.LoggerInternalMsg.Infof("Application %s is added.", applicationEvent.UUID)
 		} else if applicationEvent.Event.Type == applicationUpdate {
-			eh.AppList.List = removeApplication(eh.AppList.List, applicationEvent.ApplicationID)
-			eh.AppList.List = append(eh.AppList.List, application)
-			logger.LoggerInternalMsg.Infof("Application %s is added.", applicationEvent.ApplicationName)
+			eh.ApplicationMap[application.UUID] = &application
+			logger.LoggerInternalMsg.Infof("Application %s is updated.", applicationEvent.ApplicationName)
```

Comment:
For, create and delete applications we are logging UUID, is there any reason to log ApplicationName for update events?

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/2126/commits/618511c91e7788869966c069893c6ec4acb27919